graphql/parser.go

```diff
@@ -384,31 +390,29 @@ func Parse(source string, vars map[string]interface{}) (*Query, error) {
 	for name, fragment := range fragmentDefinitions {
 		selectionSet, err := parseSelectionSet(fragment.SelectionSet, globalFragments, vars)
 		if err != nil {
-			return nil, err
+			return rv, err
```

Comment:
Not a big fan of returning semi valid data and error at the same time, but I can't think of a simple alternative.

Reply:
somethingsomethingoptions. I think this fine, since callers should be always checking for `err` anyways (vs. `query == nil` or something).